topics/tutorial-first-steps-with-kotlin-rpc.topic

```diff
@@ -0,0 +1,599 @@
+                        tool panel:
+                    </p>
+                    <img src="tutorial_kotlin_rpc_run_server.png" alt="Run server output in intelliJ IDEA"
+                         style="block" width="706" border-effect="line"/>
+                </step>
+                <step>
+                    Navigate to the
+                    <path>Client.kt</path>
+                    file and run the application. You should see the following output in the
+                    console:
+                    <code-block lang="shell">
+                        Your pizza cost 7.89
+
+                        Process finished with exit code 0
+                    </code-block>
+                </step>
+            </procedure>
+        </chapter>
+        <p>
+            Having created a basic example, let’s now dig deeper into how everything works. In particular, let’s compare
+            and contrast Kotlin RPC with the two main alternatives – REST and gRPC.
+        </p>
+    </chapter>
+    <chapter title="Extend the example" id="extend-the-example">
+        <p>
+            Finally, let's enhance the complexity of our example application to establish a solid foundation for future
+            development.
+        </p>
+        <procedure id="extend-server">
+            <step>
+                In the
+                <path>PizzaShop.kt</path>
+                file, extend the <code>orderPizza</code> method by including the client’s ID, and add a
+                <code>viewOrders</code> method that returns all the pending orders for a specified client:
+                <code-block lang="kotlin"
+                            src="snippets/tutorial-kotlin-rpc-app/src/main/kotlin/model/PizzaShop.kt"
+                            include-lines="1-7,11-13"
+                />
+                <p>
+                    You can take advantage of the coroutines library by returning a <code>Flow</code> rather than a
+                    <code>List</code> or <code>Set</code>. This will allow you to steam the information to the client
+                    one pizza at a time.
+                </p>
+            </step>
+            <step>
+                Navigate to the
+                <path>Server.kt</path>
+                file and implement this functionality by storing the current orders in a map of lists:
+                <code-block lang="kotlin"
+                            src="snippets/tutorial-kotlin-rpc-app/src/main/kotlin/Server.kt"
+                            include-lines="18-20,26-49"
+                />
+                <p>
+                    Note that a new instance of <code>PizzaShopImpl</code> is created for each client service instance.
+                    This avoids conflicts between clients by isolating their state. However, it does not address thread safety within a single client's
+                    instance, particularly if the same instance is accessed concurrently by multiple coroutines.
```

Comment:
`it does not address thread safety within a single client's instance`

A small type, should be `within a single server's instance`, as the user state is present only on the server.

Other than that, looks great!